packages/components/src/palette-edit/index.tsx

```diff
@@ -288,15 +293,17 @@ function PaletteEditListView< T extends Color | Gradient >( {
 	useEffect( () => {
 		elementsReference.current = elements;
 	}, [ elements ] );
+
 	useEffect( () => {
 		return () => {
 			if (
-				elementsReference.current?.some( ( element ) =>
-					isTemporaryElement( slugPrefix, element )
+				elementsReference.current?.some( ( element, index ) =>
+					isDefaultElement( slugPrefix, element, index + 1 )
```

Comment:
TODO:

To make things more maintainable and predictable, we can abstract some logic from `getNameForPosition()`, the function responsible for naming the colors according to their indexed position in the array.